contrib/cell_star/core/snake.py

```diff
@@ -0,0 +1,470 @@
+
+        im = self.images.image_back_difference_blurred
+        imb = self.images.brighter
+        imc = self.images.darker
+        imfg = self.images.foreground_mask
+
+        steps = polar_transform.steps
+        max_r = polar_transform.max_r # Taka sama wartość w matlabie - generuje niezgodności !
+        R = polar_transform.R.flat
+        t = polar_transform.t
+
+        px = float(self.centroid_x) + polar_transform.x
+        px = np.maximum(px, 0)
+        px = np.minimum(px, im.shape[1] - 1)
+
+        py = float(self.centroid_y) + polar_transform.y
+        py = np.maximum(py, 0)
+        py = np.minimum(py, im.shape[0] - 1)
+
+        #
+        #
+        # Obliczanie funkcji oceny konturu i znajdowanie konturu
+        #
+        #
+
+        # indeks idzie kolejno dla każdego kąta wzdłuż promienia od najmniejszego do największego
+        # for angle:
+        #   for radius:
+        # OPT line below takes half of the total runtime of star_multi_vec
+        index = calc_util.index(px.round(), py.round()).reshape((polar_transform.x.shape[0], polar_transform.x.shape[1], 2))
+        im_index = index.reshape(polar_transform.x.size, 2)
+
+        #
+        #  pre_f - składowa funkcji oceny jakości konturu
+        #
+        #
+        #
+
+        pre_f = (cum_brightness_weight * imb[Index.to_numpy(index)] - content_weight * imc[Index.to_numpy(index)] + background_weight * (1 - imfg[Index.to_numpy(index)])) * step
+        f = np.cumsum(pre_f, axis=0)
+
+        border_thickness_steps = \
+            1 + math.floor(float(self.parameters["segmentation"]["stars"]["borderThickness"]) / float(step))
+
+        im_diff = calc_util.get_gradient(im, index, border_thickness_steps)
+
+        f_tot = f \
+            - size_weight * np.kron(np.log(R), np.ones((t.size, 1))).T \
+            - gradient_weight * im_diff \
+            - brightness_weight * im[Index.to_numpy(index)]
+
+        f_tot = f_tot.T
+
+
+        epsilon = 10**(-10)
+        f_tot = (f_tot - f_tot.min()) / (f_tot.max() - f_tot.min() + epsilon)
+        f_tot /= (np.kron(np.ones((f_tot.shape[1], 1)), f_tot.max(axis=1)).T + epsilon)
+
+        # Znajdź kontur początkowy
+        ymins, xmins = np.where(f_tot == np.kron(np.ones((f_tot.shape[1], 1)), f_tot.min(axis=1)).T)
+        # Przytnij - zabezpieczenie przed przekroczeniem dopuszczalnej liczby punktów
+        ymins, xmins = ymins[:int(points_number)], xmins[:int(points_number)]
+
+        # image_util.image_show(f_tot, 1)
+
+        #
+        #
+        # Wygładzanie konturu
+        #
+        #
+
+        # Posortuj punkty konturu (długości promieni) względem kolejnych kątów
+        s = sorted(range(len(ymins)), key=lambda k: ymins[k])
+        xmins = xmins[s]
+        # Wyznacz maksymalne różnice pomiędzy kolejnymi punktami konturu
+        max_diff = np.array(abs(smoothness) * np.arange(1, steps+1) / points_number + 0.5, dtype=int)
```

Comment:
I am getting a max_diff of zero here. Are you sure about dtype=int? I think this is where I lose all my snakes.